woocommerce/woocommerce
# Allow usage of block notice templates when using classic themes

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #44538.

This PR enables merchants to display the blocks notice templates even when using classic themes. This can be accomplished by overwriting the template files or by using the following filter:

```php
add_filter( 'woocommerce_use_block_notices_in_classic_theme', '__return_true' );
```

Apart from adjusting the display logic, I've also implemented e2e tests to ensure that this option remains working. In total, we're now having the following e2e tests regarding notice templates:

- Display **default classic notice template** when using **classic theme**.
- Display **custom classic notice template on overwrite** when using **classic theme**.
- Display **default block notice template on filter** when using **classic theme**.
- Display **custom block notice template on overwrite** when using **classic theme**.
- Display **default block notice template** when using **block theme**.
- Display **default block notice template on filter** when using **block theme**.
- Display **custom block notice template on overwrite** when using **block theme**.
- Display **custom classic notice template on overwrite** when using **block theme**.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Install the Storefront theme.
2. Install and activate the [storefront-child__block-notices-template.zip](https://github.com/woocommerce/woocommerce/files/14535269/storefront-child__block-notices-template.zip) theme.
3. Create a test page and add the following shortcode to it:

```
[woocommerce_cart]
```

4. Go to `/wp-admin/admin.php?page=wc-settings&tab=advanced` and set the test page of the previous step as the Cart page.
5. Go to `/wp-admin/edit.php?post_type=shop_coupon` and create a coupon.
6. Go to the frontend and add a product to the cart.
7. Go to the test page with the cart shortcode.
8. Provide the coupon code and verify that the block notice success template (green) is visible.
9. Provide the same coupon code again and verify that the block notice error template (red) is visible.
10. Delete the product from the cart and verify that the block notice info template (blue) is visible.
11. Activate the Storefront theme.
12. Install and activate the [Code Snippets](https://wordpress.org/plugins/code-snippets/) plugin.
13. Add the following code snippet:

```php
add_filter( 'woocommerce_use_block_notices_in_classic_theme', '__return_true' );
```

14. Repeat steps 6. until 10. and verify that the block notice templates remain visible.

Apart from the previous testing instructions, also have a look at the added and adjusted e2e tests and verify that all e2e tests are passing.

### Screenshots

<table>
<tr>
<td valign="top">Block notice success template:
<br><br>
<img width="866" alt="Screenshot 2024-02-27 at 14 05 11" src="https://github.com/woocommerce/woocommerce/assets/3323310/dbae242f-4359-462d-996f-ec5c007e9b98">
</td>
<td valign="top">Block notice error template:
<br><br>
<img width="862" alt="Screenshot 2024-02-27 at 14 05 29" src="https://github.com/woocommerce/woocommerce/assets/3323310/d3297944-8e51-4b95-94ef-099b2b7de432">
</td>
<td valign="top">Block notice info template:
<br><br>
<img width="858" alt="Screenshot 2024-02-27 at 14 05 43" src="https://github.com/woocommerce/woocommerce/assets/3323310/91a5710d-3215-4228-a448-889b264e98e4">
</td>
</tr>
</table>



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Allow usage of block notice templates when using classic themes.

#### Comment 

</details>
